Improved README.md's quality - April 17 2017

cap44 opened the pull request:
I've updated the readme file with better and cleaner language, mostly in the Windows sections because the instructions are a bit confusing. 

I've also added 2 alternative methods to build STK on Windows; slightly tweaked the OS X section and rephrased the system requirements section

this PR is mainly language updates and improvements for the README.md file

-cap44

deveee commented:
We need GL >= 3.1 to run STK and btw I don't think that disk space 400 MB is true, so you can update it too. 

And I'm not sure what about windows compilation. I think that we need one proper compilation method in readme and not two alternative methods. Maybe we should use https://github.com/supertuxkart/dependencies as our main dependencies repo (I don't think that the dependencies on sf are up to date). 

The second difference that I see is that you suggest to run msbuild from command line. This indeed should work (I was using it several times), but msbuild was not available for me by default in PATH, so I had to use full path to run it.

And if VS2017 uses different shortcuts, then maybe just at the same point with single compilation method write "since vs2017 it is ... ".

I'm fine with your other modifications.

cap44 commented:
@deveee You are correct, I mention the new OpenGL requirement here: 
>A graphics card capable of 3D rendering - NVIDIA GeForce 8 series and newer (GeForce 8100 or newer), AMD/ATI Radeon HD 4000 series and newer, Intel HD Graphics 3000 and newer. **OpenGL >= 3.3**

Not a problem, a proper method with Visual Studio will be written to replace the current one, but I do not remember the if it's just one key or a sequence of keys to build for older versions of Visual Studio. 
One more thing: I'm not sure if placing the `windows` or `windows_64bit` directories inside `stk-code` is considered a good idea.

I understand that if `MSBuild.exe` is not in the %PATH%, then the command line method gets a bit tricky. this is fine, i'll try to write a batch file that will look for MSBuild.

cap44 commented:
I modified the file slightly, because the original method for building STK on Windows is a bit confusing. 
One more thing: I've updated the disk space requirement to a minimum of  25 GB, mostly because the 64 bit version of Windows 10 requires at least 20 GB of free space.

qwertychouskie commented:
Ideally, it should be 750 MB free disk space.

qwertychouskie commented:
Otherwise, it seems like STK takes up 25 GB of space!

auriamg commented on README.md:
```diff
@@ -1,60 +1,63 @@
 * You'll need at least 512 MB of free VRAM (video memory).
-* Disk space: 400MB
-* Ideally, you want a joystick with at least 6 buttons.
+* Minimum disk space: 25 GB 
```
That is misleading, let's juste mention the approximate size STK takes on disk without attempting padding

cap44 replied:
OK, i can correct that.

auriamg commented on README.md:
```diff
@@ -1,60 +1,63 @@
   * libpng (libpng-devel)
   * zlib (zlib-devel)
   * jpeg (libjpeg-turbo-devel)
+  * vpx (libvpx-devel)
```
Benau's recording library may end up separate from STK, so maybe let's hold that one off a bit until it's clarifier how we handled video

cap44 replied:
next commit removes this line

auriamg commented on README.md:
```diff
@@ -94,22 +97,60 @@ location, specify `CMAKE_INSTALL_PREFIX` when running cmake, e.g.:
+*If CMake fails to find the dependencies during configuration; you'll need to manually add them in the missing fields*
+6. Navigate to your build directory and open the `SuperTuxKart.sln` file; Visual Studio will now load the solution. 
+7. In the 'Solution Explorer', right click on the `supertuxkart` project and select "Set as StartUp project"
+8. Open the 'Build' menu and select 'Build Solution'; or, press `CTRL + SHIFT + B` - your solution's build process has begun.
```
Let's avoid mentionning specific keybaord shortcuts because VS has various versions/profiles where keyboard shortcuts vary

auriamg commented on README.md:
```diff
@@ -94,22 +97,60 @@ location, specify `CMAKE_INSTALL_PREFIX` when running cmake, e.g.:
+5. Press 'Configure'; CMake will ask you if it is OK to create the aformentioned directory, press `Yes`. CMake will then ask you about your version of Visual Studio. 
+Confirm your selection; *Please look at the table below to avoid confusion between version numbers and releases of Visual Studio*;
+CMake will begin creating the required files for the build in the directory.
+*If CMake fails to find the dependencies during configuration; you'll need to manually add them in the missing fields*
```
I don't get that line, dependencies should be found. If they are not found then fiddling manually in cmake's variables is most likely not the right fix.

cap44 replied:
I see, the line will be removed shortly.

auriamg commented on Windows_build.md:
```diff
@@ -0,0 +1,35 @@
+----------------------|------------
+Visual Studio 2017| 15
+Visual Studio 2015| 14
+Visual Studio 2013| 13
```
Why duplicate this information in 2 files? We don't need 2 copies of it

cap44 commented:
@auriamg can you review the file once again?
I've modified the file accordingly.

On Apr 18, 2017 01:32, "auriamg" <notifications@github.com> wrote:

*@auriamg* requested changes on this pull request.
------------------------------

In README.md
<https://github.com/supertuxkart/stk-code/pull/2839#discussion_r111835320>:

>  * You'll need at least 512 MB of free VRAM (video memory).
-* Disk space: 400MB
-* Ideally, you want a joystick with at least 6 buttons.
+* Minimum disk space: 25 GB

That is misleading, let's juste mention the approximate size STK takes on
disk without attempting padding
------------------------------

In README.md
<https://github.com/supertuxkart/stk-code/pull/2839#discussion_r111835472>:

>    * Vorbis (libvorbis-dev)
   * Freetype (libfreetype6-dev)
   * libcurl (libcurl-devel)
   * libbluetooth (bluez-devel)
   * libpng (libpng-devel)
   * zlib (zlib-devel)
   * jpeg (libjpeg-turbo-devel)
+  * vpx (libvpx-devel)

Benau's recording library may end up separate from STK, so maybe let's hold
that one off a bit until it's clarifier how we handled video
------------------------------

In README.md
<https://github.com/supertuxkart/stk-code/pull/2839#discussion_r111835889>:

> +## Building SuperTuxKart on Windows
+To Build SuperTuxKart on Windows, follow these instructions:
+
+1. Download and install Visual Studio from here: [Visual Studio -
Download](https://www.visualstudio.com/downloads/). The free Visual
Studio Community edition works fine.
+2. Download the SuperTuxKart source package from either [SuperTuxKart
download area -
SourceForge.net](https://sourceforge.net/projects/supertuxkart/files/SuperTuxKart/0.9.2)
or [SuperTuxKart.net - Source
Control](https://supertuxkart.net/Source_control), and unpack it.
+*Note: If you downloaded the source package from here:
[SuperTuxKart.net - Source
Control](https://supertuxkart.net/Source_control), then both
`stk-code` and `stk-assets` **must** be in the same directory,
otherwise the build can result in failure*
+3. Download the Windows dependencies package from either
[SuperTuxKart download area: Dependecies -
SourceForge.net](https://sourceforge.net/projects/supertuxkart/files/SuperTuxKart%20Dependencies/Windows/)
+or [SuperTuxKart on GitHub -
Dependencies](https://github.com/supertuxkart/dependencies), and
unpack it; then, copy the `dependencies` directory from either the
`windows` or the `windows_64bit` directories into the `stk-code`
directory
+4. Download CMake from here: [CMake - download
page](https://cmake.org/download/), install it; once CMake is
installed, double click on the CMake icon on your desktop, and point
it towards your `stk-code` directory in the 'Where is the source code'
field, and point it to a directory called `build` or `bld`
+5. Press 'Configure'; CMake will ask you if it is OK to create the
aformentioned directory, press `Yes`. CMake will then ask you about
your version of Visual Studio.
+Confirm your selection; *Please look at the table below to avoid
confusion between version numbers and releases of Visual Studio*;
+CMake will begin creating the required files for the build in the directory.
+*If CMake fails to find the dependencies during configuration; you'll
need to manually add them in the missing fields*
+6. Navigate to your build directory and open the `SuperTuxKart.sln`
file; Visual Studio will now load the solution.
+7. In the 'Solution Explorer', right click on the `supertuxkart`
project and select "Set as StartUp project"
+8. Open the 'Build' menu and select 'Build Solution'; or, press `CTRL
+ SHIFT + B` - your solution's build process has begun.

Let's avoid mentionning specific keybaord shortcuts because VS has various
versions/profiles where keyboard shortcuts vary
------------------------------

In README.md
<https://github.com/supertuxkart/stk-code/pull/2839#discussion_r111836024>:

> -6. In Visual Studio open the project file generated in the 'bld' folder.
-7. Right click on the supertuxkart project in the solution explorer,
and select "Set as StartUp Project".
-8. Select Build->Build Solution (or press F7) to compile.
+## Building SuperTuxKart on Windows
+To Build SuperTuxKart on Windows, follow these instructions:
+
+1. Download and install Visual Studio from here: [Visual Studio -
Download](https://www.visualstudio.com/downloads/). The free Visual
Studio Community edition works fine.
+2. Download the SuperTuxKart source package from either [SuperTuxKart
download area -
SourceForge.net](https://sourceforge.net/projects/supertuxkart/files/SuperTuxKart/0.9.2)
or [SuperTuxKart.net - Source
Control](https://supertuxkart.net/Source_control), and unpack it.
+*Note: If you downloaded the source package from here:
[SuperTuxKart.net - Source
Control](https://supertuxkart.net/Source_control), then both
`stk-code` and `stk-assets` **must** be in the same directory,
otherwise the build can result in failure*
+3. Download the Windows dependencies package from either
[SuperTuxKart download area: Dependecies -
SourceForge.net](https://sourceforge.net/projects/supertuxkart/files/SuperTuxKart%20Dependencies/Windows/)
+or [SuperTuxKart on GitHub -
Dependencies](https://github.com/supertuxkart/dependencies), and
unpack it; then, copy the `dependencies` directory from either the
`windows` or the `windows_64bit` directories into the `stk-code`
directory
+4. Download CMake from here: [CMake - download
page](https://cmake.org/download/), install it; once CMake is
installed, double click on the CMake icon on your desktop, and point
it towards your `stk-code` directory in the 'Where is the source code'
field, and point it to a directory called `build` or `bld`
+5. Press 'Configure'; CMake will ask you if it is OK to create the
aformentioned directory, press `Yes`. CMake will then ask you about
your version of Visual Studio.
+Confirm your selection; *Please look at the table below to avoid
confusion between version numbers and releases of Visual Studio*;
+CMake will begin creating the required files for the build in the directory.
+*If CMake fails to find the dependencies during configuration; you'll
need to manually add them in the missing fields*

I don't get that line, dependencies should be found. If they are not found
then fiddling manually in cmake's variables is most likely not the right
fix.
------------------------------

In Windows_build.md
<https://github.com/supertuxkart/stk-code/pull/2839#discussion_r111836248>:

> +or [SuperTuxKart on GitHub - Dependencies](https://github.com/supertuxkart/dependencies), and unpack it; then copy either the `windows` or `windows_64bit` to the `stk-code` directory
+4. Download CMake from here: [CMake - download
page](https://cmake.org/download/), install it; once CMake is
installed, double click on the CMake icon on your desktop, and point
it towards your `stk-code` directory in the 'Where is the source code'
field, and point it to a directory called `build` or `bld`
+5. Press 'Configure'; CMake will ask you if it is OK to create the
aformentioned directory, press `Yes`. CMake will then ask you about
your version of Visual Studio.
+Confirm your selection; CMake will begin creating the required files
for the build in the directory.
+*If CMake fails to find the dependencies during configuration; you'll
need to manually add them in the missing fields*
+6. Navigate to your build directory and open the `SuperTuxKart.sln`
file; Visual Studio will now load the solution.
+7. In the 'Solution Explorer', right click on the `supertuxkart`
project and select "Set as StartUp project"
+8. Open the 'Build' menu and select 'Build Solution'; or, press `CTRL
+ SHIFT + B` - your solution's build process has begun.
+
+*Note: To avoid confusion between releases and versions, refer to this table:*
+
+Visual Studio Release | Version
+----------------------|------------
+Visual Studio 2017| 15
+Visual Studio 2015| 14
+Visual Studio 2013| 13

Why duplicate this information in 2 files? We don't need 2 copies of it

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/supertuxkart/stk-code/pull/2839#pullrequestreview-33105090>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/AHpV-eb9hdn2YNcT4Nb7c2NbzJsAxRojks5rw-hsgaJpZM4M-ws7>
.

deveee commented:
You didn't correct the OpenGL >= 3.3. We need OpenGL >= 3.1 (and intel HD3000 on windows has only GL 3.1). I was rather long time ago when I was testing it on linux, but it was working too. If it doesn't work for you with GL 3.1, then report a bug :)

auriamg commented:
@deveee I've fixed the OpenGL version
